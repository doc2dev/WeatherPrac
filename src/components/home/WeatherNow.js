import React, { Component } from 'react'
import { View, Text, Animated } from 'react-native'
import styles from '../../styles/home'
import { titleCase } from '../../util'

export default class WeatherNow extends Component {
  render() {
    const { data, scrollY } = this.props
    const windowHeight = 300
    const weatherTitleCase = titleCase(data.weather)
    const tempTextStyle = {
      opacity: scrollY.interpolate({
          inputRange: [-windowHeight, 0, windowHeight],
          outputRange: [1, 1, 0]
      })
    }
    const collapseStyle = [
      styles.weatherToday,
      {
        height: scrollY.interpolate({
            inputRange: [-windowHeight, 0, windowHeight / 1.2],
            outputRange: [180, 150, 80]
        })
      }
    ]
    return(
      <Animated.View style={collapseStyle}>
        <Text style={styles.weatherNowCity}>{data.place}</Text>
        <Text style={styles.weatherNowDesc}>{weatherTitleCase}</Text>
        <Animated.View style={tempTextStyle}>
          <Text style={styles.weatherNowTemp}>{data.temperature}</Text>
        </Animated.View>
      </Animated.View>
    )
  }
}
