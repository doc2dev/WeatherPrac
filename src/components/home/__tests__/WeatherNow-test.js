import 'react-native';
import React from 'react';
import WeatherNow from '../WeatherNow';
import renderer from 'react-test-renderer';
import data from './data'
import sinon from 'sinon'

describe('Weather now: ', () => {

  const mock = {
    place: 'TestPlace',
    weather: 'snowing hard',
    temperature: '23',
    todaysWeather: data
  }
  const scrollY = {
    interpolate: sinon.stub().returns(42)
  }
  const tree = renderer
    .create(<WeatherNow data={mock} scrollY={scrollY}/>)
    .toJSON();

  it('renders correctly', () => {
    /*console.log('Test tree object: ', tree)*/
    /*console.log('Expect object: ', expect(tree))*/
    expect(tree).toMatchSnapshot();
  });

  it('should display place name', () => {
    const placeView =tree.children[0]
    expect(placeView.type).toBe('Text')
    expect(placeView.children[0]).toBe(mock.place)
  })

  it('should display weather description', () => {
    const weatherView = tree.children[1]
    expect(weatherView.type).toBe('Text')
    expect(weatherView.children[0]).toBe('Snowing Hard')
  })

  it('should display temperature text', () => {
    const tempText = tree.children[2].children[0]
    expect(tempText.type).toBe('Text')
    expect(tempText.children[0]).toBe(mock.temperature)
  })
})
