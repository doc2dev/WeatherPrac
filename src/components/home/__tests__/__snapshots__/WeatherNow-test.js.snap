// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather now:  renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 42,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "sans-serif-medium",
        "fontSize": 24,
      }
    }
  >
    TestPlace
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    Snowing Hard
  </Text>
  <View
    style={
      Object {
        "opacity": 42,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "sans-serif-light",
          "fontSize": 40,
        }
      }
    >
      23
    </Text>
  </View>
</View>
`;
